server/src/test/java/org/elasticsearch/snapshots/SnapshotsServiceTests.java

```diff
@@ -191,7 +213,147 @@ public void testSuccessfulSnapshot() {
         assertEquals(0, snapshotInfo.failedShards());
     }
 
-    private static <T> ActionListener<T> assertingListener(Runnable r) {
+    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/36779")
+    public void testMasterDisconnectDuringSnapshot() {
```

Comment:
This test is capable of reproducing the exact scenario described here https://github.com/elastic/elasticsearch/issues/36779#issuecomment-450665392